test/evergreen.yml

```diff
@@ -1435,6 +1435,15 @@ tasks:
             set -o errexit
             set -o verbose
             ./configure_combinations.sh ${smp_command|} 2>&1
+      # Handle special build combination for running all the diagnostic tests.
+      - func: "configure wiredtiger"
+        vars:
+          configure_env_vars: CFLAGS="-g -Werror"
+          posix_configure_flags: --enable-silent-rules --enable-diagnostic --disable-static
+      - func: "make wiredtiger"
+        vars:
+          make_command: make all
+      - func: "make check all"
```

Comment:
Splitting the `make check all` part out is one solution - if the previous `configure_combinations.sh` test fails, there's risk the build_posix directory is to be overwritten by the later funcs added here, which means we lose environment to troubleshoot the test failure. That's actually why I prefer separate compile and make check tasks instead of separate funcs within the same task. 

The task is an unit in Evergreen that has its own artifacts archived.

Reply:
I just ran a mock test where the `configure_combination.sh` script returned 1. The task was aborted immediately and no further commands were executed from the task. Given this built-in safety mechanism, I believe it is safe to just add functions sequentially with a guarantee that no command/function will be executed if any of the previous command/function fails.